lib/services/local/users.go

```diff
@@ -563,12 +563,15 @@ func (s *IdentityService) CompareAndSwapUser(ctx context.Context, new, existing
 
 		item.Revision = currentWithoutSecrets.GetRevision()
 
-		if _, err = s.Backend.ConditionalUpdate(ctx, item); err != nil {
+		lease, err := s.Backend.ConditionalUpdate(ctx, item)
+		if err != nil {
 			if trace.IsCompareFailed(err) {
 				continue
 			}
 			return trace.Wrap(err)
 		}
+		new.SetRevision(lease.Revision)
```

Comment:
Does this actually matter? Don't we clear the revision before writing and after reading from the backend?